### PR TITLE
FND-298 - Several changes are needed to the definition of numeric index

### PR DESCRIPTION
**Pull request by ElisaKendall**
Signed-off-by: Elisa Kendall <ekendall@thematix.com>

## Description

Revised the definition of numeric index to be numeric index value per discussion in the IND FCT; updated its use in economic indicators accordingly

Fixes: #835 / FND-298

## Checklist:

- [x] I'm familiar with the [FIBO developer quide](../CONTRIBUTING.md#contributing-to-the-fibo-code). My contribution meets all the requirements described there.
- [x] My contribution follows the [principles of best practices for FIBO](../ONTOLOGY_GUIDE.md).
- [x] My changes have been reconciled with latest master and no merge conflicts remain.
- [x] This PR is related to exactly one issue. The issue is referenced by using a GitHub keyword such as "fixes", "closes", or "resolves".
- [x] Hygiene tests have been applied by a PR with "(WIP)" in title.
- [x] The issue has been tested locally using a reasoner (for ontology changes).

### CODE REVIEW

File: FND/Utilities/Analytics.rdf [hunk 1]
```diff
@@ -658,6 +674,13 @@ For certain indices, one of the most common weighting factor is by market capita
 		<fibo-fnd-utl-av:synonym>has independent variable</fibo-fnd-utl-av:synonym>
 	</owl:ObjectProperty>
 	
+	<owl:ObjectProperty rdf:about="&fibo-fnd-utl-alx;hasBaseDate">
+		<rdfs:subPropertyOf rdf:resource="&fibo-fnd-dt-fd;hasExplicitDate"/>
+		<rdfs:label>has base date</rdfs:label>
+		<rdfs:range rdf:resource="&fibo-fnd-dt-fd;ExplicitDate"/>
+		<skos:definition>specifies the base date against which the value of a numberic index for a more recent date is compared (or from which it stems)</skos:definition>
```
Comment (rivettp) [inline FND/Utilities/Analytics.rdf:681]: typo "numberic"

File: FND/Utilities/Analytics.rdf [hunk 2]
```diff
@@ -397,6 +399,20 @@
 				<owl:allValuesFrom rdf:resource="&fibo-fnd-dt-fd;RecurrenceInterval"/>
 			</owl:Restriction>
 		</rdfs:subClassOf>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-fnd-utl-alx;hasBaseDate"/>
```
Comment (rivettp) [inline FND/Utilities/Analytics.rdf:404]: should have an explanatoryNote that only one of baseDate and basePeriod are expected to be used

File: FND/Utilities/Analytics.rdf [hunk 3]
```diff
@@ -305,25 +308,24 @@
...
+	<owl:Class rdf:about="&fibo-fnd-utl-alx;NumericIndexValue">
+		<rdfs:subClassOf rdf:resource="&fibo-fnd-qt-qtu;QuantityValue"/>
 		<rdfs:subClassOf>
 			<owl:Restriction>
-				<owl:onProperty rdf:resource="&fibo-fnd-arr-doc;hasReportingPeriod"/>
-				<owl:onClass rdf:resource="&fibo-fnd-dt-fd;ExplicitDatePeriod"/>
-				<owl:qualifiedCardinality rdf:datatype="&xsd;nonNegativeInteger">1</owl:qualifiedCardinality>
+				<owl:onProperty rdf:resource="&fibo-fnd-utl-alx;isMeasureOf"/>
+				<owl:onClass rdf:resource="&fibo-fnd-utl-alx;ScopedMeasure"/>
+				<owl:minQualifiedCardinality rdf:datatype="&xsd;nonNegativeInteger">0</owl:minQualifiedCardinality>
```
Comment (rivettp) [inline FND/Utilities/Analytics.rdf:317]: Why make this optional? When would this not be set - without the Measure then the value has no context/meaning

File: FND/Utilities/Analytics.rdf [hunk 4]
```diff
@@ -738,6 +761,12 @@ For certain indices, one of the most common weighting factor is by market capita
 		<skos:definition>indicates the number of elements in a given population</skos:definition>
 	</owl:DatatypeProperty>
 	
+	<owl:DatatypeProperty rdf:about="&fibo-fnd-utl-alx;hasReferenceDateTime">
```
Comment (rivettp) [inline FND/Utilities/Analytics.rdf:764]: not sure about "Reference" here which could get confused with "Base".

File: FND/Utilities/Analytics.rdf [hunk 5]
```diff
@@ -738,6 +761,12 @@ For certain indices, one of the most common weighting factor is by market capita
...
+		<rdfs:label>has reference date and time</rdfs:label>
+		<rdfs:range rdf:resource="&fibo-fnd-dt-fd;CombinedDateTime"/>
+		<skos:definition>specifies a reference date and possibly a time for a given value</skos:definition>
```
Comment (rivettp) [inline FND/Utilities/Analytics.rdf:767]: Circular definition. 
Also please avoid "specifies" in definitions. 
Need to clarify that's it's the time when the value was valid, as opposed to the time it was measured/sampled or published.
Finally I think this should be "the" not "a".

File: IND/EconomicIndicators/EconomicIndicators.rdf [hunk 6]
```diff
@@ -389,7 +390,7 @@
 		<rdfs:subClassOf>
 			<owl:Restriction>
 				<owl:onProperty rdf:resource="&fibo-ind-ei-ei;hasIndicatorValue"/>
-				<owl:onClass rdf:resource="&fibo-fnd-utl-alx;NumericIndex"/>
+				<owl:onClass rdf:resource="&fibo-fnd-utl-alx;NumericIndexValue"/>
```
Comment (rivettp) [inline IND/EconomicIndicators/EconomicIndicators.rdf:393]: This seems wrong! The CPI for the US, for example, has many values, each for a different date. Is this meant to be the latest value - in which case the property name should indicate that e.g. hasLatestValue. 
In general case I think the reference should be from the values to the index (which you have), not vice versa.

File: FND/Utilities/Analytics.rdf (hunk 1 above)
Comment (ElisaKendall) [reply FND/Utilities/Analytics.rdf:681]: fixed

File: FND/Utilities/Analytics.rdf (hunk 2 above)
Comment (ElisaKendall) [reply FND/Utilities/Analytics.rdf:404]: The FCT changed the names of these properties and added two additional properties to be clearer, but we need to review how they are applied.  hasBaseDate is now hasAnchorDate, which is the start for a series, to be clear about what that means.  We added hasFixedComparativeDate (e.g., the end of the last recession), hasRelativeComparativeDate (e.g. prior reported figure), and changed hasBasePeriod to hasRelativeComparativePeriod, again for clarity, but we need further analysis to determine whether they belong on scoped measure or something else.

File: FND/Utilities/Analytics.rdf (hunk 3 above)
Comment (ElisaKendall) [reply FND/Utilities/Analytics.rdf:317]: Still under discussion - it's possible that not all analyses that use scoped measures have an anchor date - maybe they do, and they certainly do for economic indicators, but scoped measure is broader than that.

File: FND/Utilities/Analytics.rdf (hunk 5 above)
Comment (ElisaKendall) [reply FND/Utilities/Analytics.rdf:767]: Not sure what you are thinking here - the name of the class this restriction is on is NumericIndexValue, whose definition is "change in some aggregate relative to the value of the aggregate as of some reference date" - the property definition does not say anything about when it was measured or published.  We could delete this altogether and just use hasObservedDateTime if you prefer.

File: IND/EconomicIndicators/EconomicIndicators.rdf (hunk 6 above)
Comment (ElisaKendall) [reply IND/EconomicIndicators/EconomicIndicators.rdf:393]: fixed

File: FND/Utilities/Analytics.rdf (hunk 4 above)
Comment (ElisaKendall) [reply FND/Utilities/Analytics.rdf:758]: See above on the renaming discussion.

File: FND/Utilities/Analytics.rdf (hunk 3 above)
Comment (rivettp) [reply FND/Utilities/Analytics.rdf:317]: My comment was about isMeasureOf not any date. When would a IndexValue not be linked to a Measure. 
BTW the property name seems wrong now I've looked at it again - it should be isValueOf not isMeasureOf.

File: FND/Utilities/Analytics.rdf (hunk 5 above)
Comment (rivettp) [reply FND/Utilities/Analytics.rdf:761]: we seem to be at cross purposes. My comment was on property hasReferenceDateAndTime and the fact that its definition "specifies a reference date...for a given value" was circular. The definition should be something like "the time when the value actually occurred"

File: IND/EconomicIndicators/EconomicIndicators.rdf (hunk 6 above)
Comment (rivettp) [reply IND/EconomicIndicators/EconomicIndicators.rdf]: You didn't address my 2nd point which is that the direction should be that the value should reference the Index; we should not encourage people to try to link from the Index to reference all its values over time.

Comment (ElisaKendall) [reply IND/EconomicIndicators/EconomicIndicators.rdf]: Let's do this in another issue - I think more work may be involved to clean some of this up.

File: FND/Utilities/Analytics.rdf (hunk 3 above)
Comment (rivettp) [reply FND/Utilities/Analytics.rdf:317]: Still think isValueOf (good name change) should have cardinality of 1.

File: IND/EconomicIndicators/EconomicIndicators.rdf [hunk 7]
```diff
@@ -388,28 +386,26 @@
...
+				<owl:onProperty rdf:resource="&fibo-fnd-utl-alx;hasArgument"/>
 				<owl:someValuesFrom rdf:resource="&fibo-ind-ei-ei;CivilianNonInstitutionalPopulation"/>
 			</owl:Restriction>
 		</rdfs:subClassOf>
 		<rdfs:subClassOf>
 			<owl:Restriction>
-				<owl:onProperty rdf:resource="&fibo-fnd-utl-alx;hasArgument"/>
-				<owl:someValuesFrom rdf:resource="&fibo-fbc-pas-fpas;Basket"/>
+				<owl:onProperty rdf:resource="&fibo-ind-ei-ei;hasIndicatorValue"/>
+				<owl:someValuesFrom rdf:resource="&fibo-fnd-utl-alx;NumericIndexValue"/>
```
Comment (rivettp) [inline IND/EconomicIndicators/EconomicIndicators.rdf:408]: As discussed I think it not a good idea to, not only encourage but, require triples to be established in this direction (from Indicator to potentially hundreds of values).

**Pull request merged**